.github/workflows/build.yml

```diff
@@ -461,44 +461,77 @@ jobs:
-          CWD=`pwd`
           cd pr-branch
           ../ve1/bin/chart-repo-manager \
             --repository=${{ github.repository }} \
-            --index-branch=${INDEX_BRANCH} \
             --api-url=${{ github.event.pull_request._links.self.href }} \
-            --pr-number=${{ github.event.number }}
-          cd ${CWD}
 
-      - name: Release
+      # Upload the report file, potentially paired with a public key and, if provided, the chart's tarball and its prov file.
+      # Only the report file is always included.
+      # The release tag format is <organization_name>-<chart_name>-<chart_version>
+      - name: Create GitHub release
         if: ${{ needs.chart-verifier.outputs.web_catalog_only == 'False' }}
-        uses: softprops/action-gh-release@v0.1.12
-        continue-on-error: true
+        uses: softprops/action-gh-release@v0.1.15
         with:
           tag_name: ${{ needs.chart-verifier.outputs.release_tag }}
           files: |
-              ${{ steps.release-charts.outputs.report_file }}
-              ${{ steps.release-charts.outputs.public_key_file }}
+              ${{ steps.prepare-chart-release.outputs.report_file }}
+              ${{ steps.prepare-chart-release.outputs.public_key_file }}
+              ${{ steps.prepare-chart-release.outputs.path_to_chart_tarball }}
+              ${{ steps.prepare-chart-release.outputs.prov_file_name }}
           fail_on_unmatched_files: true
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Update Helm repository index
+        if: ${{ needs.setup.outputs.run_build == 'true' }}
+        env:
+          CHART_ENTRY_NAME: ${{ needs.chart-verifier.outputs.chart_entry_name }}
+          WEB_CATALOG_ONLY: ${{ needs.chart-verifier.outputs.web_catalog_only }}
+        run: |
+          INDEX_BRANCH=$(if [ "${GITHUB_REF}" = "refs/heads/main" ]; then echo "gh-pages"; else echo "${GITHUB_REF##*/}-gh-pages"; fi)
+
+          echo "[INFO] Creating Git worktree for index branch"
+          INDEX_DIR=$(mktemp -d)
+          git remote add upstream "$GITHUB_SERVER_URL"/"$GITHUB_REPOSITORY"
+          git fetch upstream "$INDEX_BRANCH"
+          git worktree add --detach "$INDEX_DIR" upstream/"$INDEX_BRANCH"
+
+          if [ "$WEB_CATALOG_ONLY" == "True" ]; then
+            INDEX_FILE="unpublished-certified-charts.yaml"
+          else
+            INDEX_FILE="index.yaml"
+          fi
+
+          source ve1/bin/activate
+          cd $INDEX_DIR
+          update-index \
+            --index-branch=${INDEX_BRANCH} \
+            --index-file=${INDEX_FILE} \
+            --repository=${{ github.repository }} \
+            --chart-entry='${{ steps.prepare-chart-release.outputs.chart_entry }}' \
+            --chart-url="${{ steps.prepare-chart-release.outputs.chart_url }}" \
+            --version="${{ steps.prepare-chart-release.outputs.version }}"
```

Comment:
This is one of those CI-isms - should these values be inputs to the step (e.g. as `env` definitions) or called directly here?

Making these inputs to the step makes it a bit more portable, and I do think this step will benefit from portability, given that I'd like to be able to re-run index modification work in the future in cases where steps have failed transiently.

Reply:
So far I considered a bad practice to extensively use `os.environ.get` within our python script to pass what essentially are simple arguments for this scripts. Sometimes those calls to `os.environ.get` are made deep inside some function calls, which can make it really cryptic what environment variables are actually used and where.

Hence, when creating this new `update-index` script (though it's mostly code shifted away from `chart-repo-manager`), I avoided using environment variables and preferred `argparse`.

But... it seems you're not actually commenting on this, rather you're suggesting to have an environment variable, that is then used as argument to the python script ? At the moment I fail to see how that makes the step more "portable", but I'm happy to move to this kind of design if you prefer.

Reply:
Understood. I think it makes sense.

I will Resolve for now and keep this in mind for future iterations. It's unfortunately not that trivial here because `${{ steps.prepare-chart-release.outputs.chart_entry }}` holds a dict that is loaded using `json.loads` in `argparse`, and it needs to be single-quoted in order to work. But if I use a environment variable it won't be evaluated because of the single quotes.

```python
    parser.add_argument(
        "-e",
        "--chart-entry",
        dest="chart_entry",
        type=json.loads,
        required=True,
        help="Index entry to add",
    )
```

It may not be a very good idea. It's actually the first time I use `json.loads` in as a argparse type. I could write to a file, or use base64 encoding/decoding instead... As we don't consider this blocking I'll keep this for a later iteration.